packages/web/src/js/Collapse.ts

```diff
@@ -0,0 +1,167 @@
+import BaseComponent from './BaseComponent';
+import { enableToggleAutoloader } from './utils/ComponentFunctions';
+import selectorEngine from './dom/SelectorEngine';
+import EventHandler from './dom/EventHandler';
+
+const NAME = 'collapse';
+const NAME_ARIA_EXPANDED = 'aria-expanded';
```

Comment:
There is also missing `DATA_KEY`, used by BaseComponent.

Also please add `show`, `shown`, `hide`, `hidden` events like in https://github.com/twbs/bootstrap/blob/main/js/src/collapse.js#L28. We want to have an underlying event API which developers can use to attach some other functionality to Collapse actions.